Source/JavaScriptCore/runtime/JSString.cpp

```diff
@@ -198,9 +198,9 @@ AtomString JSRopeString::resolveRopeToAtomString(JSGlobalObject* globalObject) c
 
     if (length() > maxLengthForOnStackResolve) {
         scope.release();
-        return resolveRopeWithFunction(globalObject, [&] (Ref<StringImpl>&& newImpl) {
+        return AtomString { resolveRopeWithFunction(globalObject, [&] (Ref<StringImpl>&& newImpl) {
```

Comment:
I think we could change this to not need to check the atom string bit at runtime, since we know this is an `AtomStringImpl`.

Reply:
Ok, adding a FIXME comment for now.